File: app/src/main/java/com/gmail/laktionov/sample/rx/lyricsfinder/search/datasource/remote/RemoteSource.java
package com.gmail.laktionov.sample.rx.lyricsfinder.search.datasource.remote;

import com.gmail.laktionov.sample.rx.lyricsfinder.search.datasource.remote.model.SearchResponse;

import io.reactivex.Maybe;

public interface RemoteSource {
    Maybe<SearchResponse> searchLyric(String artistName, String artistSong);
}

File: app/src/main/java/com/gmail/laktionov/sample/rx/lyricsfinder/core/datasource/RemoteSource.kt
package com.gmail.laktionov.sample.rx.lyricsfinder.core.datasource

import com.gmail.laktionov.sample.rx.lyricsfinder.core.datasource.remote.LyricRequest
import com.gmail.laktionov.sample.rx.lyricsfinder.core.datasource.remote.LyricResponse

interface RemoteSource {

    fun findLyricRemote(request: LyricRequest): LyricResponse
}